hedera-node/hedera-mono-service/src/main/java/com/hedera/services/state/merkle/MerkleToken.java

```diff
@@ -707,6 +718,67 @@ public List<CustomFee> grpcFeeSchedule() {
         return grpcList;
     }
 
+    public Optional<TokenInfo> asTokenInfo(final TokenID tokenId, final ByteString ledgerId) {
```

Comment:
I really like this idea to have the token provide a new view of itself; but we probably don't need to return an `Optional<TokenInfo>`, since if we can call this method, the token must exist. 😁

Reply:
DONE, I agree with you :)